src/Symfony/Component/Security/Core/Encryption/SymmetricEncryptionInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encryption;
+
+use Symfony\Component\Security\Core\Exception\EncryptionException;
+use Symfony\Component\Security\Core\Exception\WrongEncryptionKeyException;
+
+/**
+ * Symmetric encryption uses the same key to encrypt and decrypt a message. The
+ * keys should be kept safe and should not be exposed to the public. Symmetric
+ * encryption should be used when you are not sending the encrypted message to
+ * anyone else.
+ *
+ * Example: You store a value on disk and don't want anyone else to read it.
+ *
+ * Symmetric encryption is in theory weaker than asymmetric encryption.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental in 5.2
+ */
+interface SymmetricEncryptionInterface
+{
+    /**
+     * Get an encrypted version of the message.
+     *
+     * @param string $message plain text version of the message
+     *
+     * @throws EncryptionException
+     */
+    public function encrypt(string $message): string;
```

Comment:
It could be strange that we don't define a `$key` argument here. We are forcing all implementations of `SymmetricEncryptionInterface` to accept a key in the constructor.

Reply:
That's a design decision we might discuss, but why not do the same for the asymetric encryption?

Reply:
Here is how I've been reasoning with myself. I might be wrong so just disagree with me if you want. 

**Symmetric**: This is used when the same application encrypts and decrypts the message. Ie, storing a value on disk, or when you send a user a magic login link, you want to make sure the query parameters has not been tampered with. Any situation where you want to hide a value. Like `sha1()` but also reversible. 

There is no need for you to have multiple keys. It will never be more secure. You are also never sharing the key, there is no such strategy like "if I have multiple keys, then if one get exposed, all my secrets aren't exposed". (Either none of your key is exposed, or all of them)

That is why the `$key` is not a runtime config, but something you configure in build time. 

**Asymmetric**:
This is about encrypting data to send to other computers. It is also about validating signatures. If you want to send data two 5 computers, you don't want to instantiate 5 different objects if `AsymmetricEncryptionInterface`. 

The Asymmetric keys are also likely to be stored in a database/datastore.